spec/contacts/removing_a_contact_spec.rb

```diff
@@ -0,0 +1,43 @@
+feature "Removing a contact", contacts: true, finder_frontend: true, government_frontend: true, new: true do
+  include ContactsHelpers
+
+  let(:title) { "Removing a contact #{SecureRandom.uuid}" }
+
+  scenario "Removing a contact" do
+    given_there_is_a_published_contact
+    when_i_remove_the_contact
+    then_i_get_a_404_on_gov_uk
+    and_it_is_not_shown_on_finder
+  end
+
+  private
+
+  def given_there_is_a_published_contact
+    publish_contact(title: title)
+    @url = find_link(title)[:href]
+    reload_url_until_status_code(@url, 200)
```

Comment:
Not sure we need to loop here. I reckon since we're expecting a 410 it'd be ok if the later loop was for: `reload_url_until_status_code(@url, 410, keep_retrying_while: [404, 200])`

Reply:
Our previous scenarios have suffered from race conditions where the unpublish hit the api before the publish for example [here](https://github.com/alphagov/publishing-e2e-tests/pull/113)

Reply:
I think that instance is a special case which highlights an issue in the content store app.

When Publishing API communicates with Content Store there's the payload version to ensure that it accepts newer updates and rejects older ones (there's also a [db lock](https://github.com/alphagov/publishing-api/blob/1b099acb414b1bd680ee63dbc45e7373bed29179/app/clients/content_store_writer.rb) so that same base_paths don't write at same time)

However in the case of discard draft, it deletes something from the content store so it loses that context to check the item is up to date, which can make an out of order execution cause an inconsistent state. 

So I think the race condition issue should only be a concern if we're dealing with a delete request going to content store which doesn't happen with unpublishings (instead update item with a redirect or gone).

I should try find some time to think about fixing that issue with content store too :-/

Reply:
Makes sense.  I'll remove it from the Whitehall one.